plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx

```diff
@@ -121,8 +137,179 @@ export const Edit = ( {}: BlockEditProps< ProductFiltersBlockAttributes > ) => {
+							<ExternalLink href={ templatePartEditUri }>
+								{ __( 'Edit overlay', 'woocommerce' ) }
+							</ExternalLink>
+						</p>
+					) }
+					{ ( attributes.overlay === 'mobile' ||
+						attributes.overlay === 'always' ) && (
+						<>
+							<RadioControl
+								className="wc-block-editor-product-filters__overlay-button-style-toggle"
+								label={ __( 'BUTTON', 'woocommerce' ) }
+								selected={ attributes.overlayButtonStyle }
+								onChange={ (
+									value: 'label-icon' | 'label' | 'icon'
+								) => {
+									setAttributes( {
+										overlayButtonStyle: value,
+									} );
+								} }
+								options={ [
+									{
+										value: 'label-icon',
+										label: __(
+											'Label and icon',
+											'woocommerce'
+										),
+									},
+									{
+										value: 'label',
+										label: __(
+											'Label only',
+											'woocommerce'
+										),
+									},
+									{
+										value: 'icon',
+										label: __( 'Icon only', 'woocommerce' ),
+									},
+								] }
+							/>
+							{ attributes.overlayButtonStyle !== 'label' && (
+								<>
+									<ToggleGroupControl
+										className="wc-block-editor-product-filters__overlay-button-toggle"
+										isBlock={ true }
+										value={ attributes.overlayIcon }
+										onChange={ (
+											value:
+												| 'filter-icon-1'
+												| 'filter-icon-2'
+												| 'filter-icon-3'
+												| 'filter-icon-4'
+										) => {
+											setAttributes( {
+												overlayIcon: value,
+											} );
+										} }
+									>
+										<ToggleGroupControlOption
+											value={ 'filter-icon-1' }
+											label={
+												<Icon
+													size={ 32 }
+													icon={
+														<svg
+															xmlns="http://www.w3.org/2000/svg"
+															viewBox="0 0 24 24"
+															width="32"
+															height="32"
+															aria-hidden="true"
+														>
+															<path d="M10.541 4.20007H5.20245C4.27908 4.20007 3.84904 5.34461 4.54394 5.95265L10.541 11.2001V16.2001L10.541 17.9428C10.541 18.1042 10.619 18.2558 10.7504 18.3496L13.2504 20.1353C13.5813 20.3717 14.041 20.1352 14.041 19.7285V11.2001L19.3339 5.90718C19.9639 5.27722 19.5177 4.20007 18.6268 4.20007H13.041H10.541Z" />
+														</svg>
+													}
+												/>
+											}
+										/>
+										<ToggleGroupControlOption
+											value={ 'filter-icon-2' }
+											label={
+												<Icon
+													size={ 32 }
+													icon={
+														<svg
+															xmlns="http://www.w3.org/2000/svg"
+															viewBox="0 0 24 24"
+															width="32"
+															height="32"
+															aria-hidden="true"
+														>
+															<path
+																d="M4 11.5H8V10H4V11.5ZM0 0V1.5H12V0H0ZM2 6.5H10V5H2V6.5Z"
+																transform="translate(6,6)"
+															/>
+														</svg>
```

Comment:
So I have a dilemma here because I am finding several locations of where icons live. Some are here https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/assets/images/blocks
Some are within the block directories themselves and some are here https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-blocks/assets/js/icons/library

Also what is the naming convention we want? We can't just name an icon file filter.tsx because there could be many different filter icons that are used with different blocks. So we need to prefix (namespace) that right? But what is the convention?

I am happy to go with your suggestion here since you've been working in this for longer.

Reply:
> Some are here https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/assets/images/blocks

This is place for images rather than icon, even images there are svg.

> Some are within the block directories themselves and some are here https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-blocks/assets/js/icons/library

Let's move all icon to this folder and follow the naming convention there. We don't need a namespace because we import them from our own alias, so there won't be any conflict. We should move existing local icons too (but in other PRs).